decoding_morse.rb

```diff
@@ -0,0 +1,33 @@
+# Create a method to decode a Morse code character
+def decode_char(morse_code)
+  morse_code_dict = {
+    '.-' => 'A',
+    '-...' => 'B',
+    '-.-.' => 'C', '-..' => 'D',
+    '.' => 'E', '..-.' => 'F', '--.' => 'G', '....' => 'H',
+    '..' => 'I', '.---' => 'J', '-.-' => 'K', '.-..' => 'L',
+    '--' => 'M', '-.' => 'N', '---' => 'O', '.--.' => 'P',
+    '--.-' => 'Q', '.-.' => 'R', '...' => 'S', '-' => 'T',
+    '..-' => 'U', '...-' => 'V', '.--' => 'W', '-..-' => 'X',
+    '-.--' => 'Y', '--..' => 'Z'
+  }
+
+  morse_code_dict[morse_code.strip] || 'UNKNOWN'
+end
+
+# Create a method to decode an entire word in Morse code
+def decode_word(morse_code)
+  morse_code.split.map { |morse_code_char| decode_char(morse_code_char) }.join
+end
+
+# Create a method to decode the entire message in Morse code
+def decode_message(morse_message)
+  words = morse_message.split('   ')
+  decoded_words = words.map do |word|
+    characters = word.chars
+    decoded_characters = characters.map { |char| decode_char(char) }
+    decoded_characters.join
+  end
+
+  decoded_words.join('   ')
+end
```

Comment:
- [ ] Good job creating the `decode_char` and the `decode_word` separately :clap: However the `decode_message` function is not returning the correct message.
  This `decode_message("-- -.--   -. .- -- .")`  supposed to returns **`"MY NAME"`** but we got this **`"TTUNKNOWNTETT   TEUNKNOWNETUNKNOWNTTUNKNOWNE"`**

  **HIT:  We can simply use the `decode_word` function in our `decode_message` function. So the `decode_message` could looks this :point_down:** (Your code doesn't need to be exactly like this)

```ruby
def decode_message(morse_message)
  morse_message.split('   ').map { |w| decode_word(w) }.join(' ')
end
```